pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+
+        Parameters
+        ----------
+        column: `TimeResponseColumn`
+            Parameter to return.
+
+        Returns
+        -------
+        list
+            The values for the requested time response column (e.g., step, pulse, etc.).
+        """
+        size = c_int()
+        status = self._dll.getIdealTimeResponseSize(byref(size))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        array = (c_double * size.value)()
+        status = self._dll.getIdealTimeResponse(array, size.value, column.value)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        values = [float(val) for val in array]
+        return values
+
+    def _sparamaters_response_getter(self, column: SParametersResponseColumn):
+        """Return the ideal filter S parameters response.
+
+        Parameters
+        ----------
+        column: `SParametersResponseColumn`
+            Parameter to return.
+
+        Returns
+        -------
+        list
+            The values for the requested S parameters response column (e.g., S11, S21, etc.).
+        """
+        size = c_int()
+        status = self._dll.getIdealSParamatersResponseSize(byref(size))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        array = (c_double * size.value)()
+        status = self._dll.getIdealSParamatersResponse(array, size.value, column.value)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        values = [float(val) for val in array]
+        return values
+
+    def _pole_zeros_response_getter(self, column: PoleZerosResponseColumn):
+        """Return ideal pole zero location parameters (e.g., x coordinate of
+        the denominator of transmission zeros, etc.).
+
+        Parameters
+        ----------
+        column: `PoleZerosResponseColumn`
+            Parameter to return.
+
+        Returns
+        -------
+        list
+            The values for the requested pole zero column.
+        """
+        size = c_int()
+        status = self._dll.getIdealPoleZerosResponseSize(byref(size), column.value)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        array = (c_double * size.value)()
+        status = self._dll.getIdealPoleZerosResponse(array, size.value, column.value)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        values = [float(val) for val in array]
+        return values
+
+    def transfer_function_response(self):
+        """Return ideal filter transfer function parameters.
+
+        Returns
+        -------
+        str
+            The requested parameter array.
+
+
+            str
+                Multi-line string where each line contains a coefficient from
+                the numerator and/or the denominator of the transfer function.
+                The coefficient for the highest-order term is first and the terms are in decreasing order.
```

Comment:
```suggestion
        """Get the ideal filter transfer function's parameters.

        Returns
        -------
        str
            Requested parameter array.


            str
                Multi-line string where each line contains a coefficient from
                the numerator and/or the denominator of the transfer function.
                The coefficient for the highest-order term is first, and the terms are in decreasing order.
```
In the "Returns" section, is the second string describing the parameter array? It wasn't easy for me to tell if the indentation was intentional. Perhaps you can make this explanation clearer?